main/management/commands/updatecomponents.py

```diff
@@ -0,0 +1,100 @@
+from django.core.management import BaseCommand
+import urllib, json
+from main.models import *
+from biojs.config import *
+
+def get_github_data(github_url):
+    response = urllib.urlopen(github_url)       # 'https://api.github.com/users/whatever?client_id=xxxx&client_secret=yyyy' format
+    data = json.loads(response.read())
+    return data
+
+def get_npm_data():
+    response = urllib.urlopen("https://api.npms.io/v2/search?q=keywords:biojs")
```

Comment:
Probably could merge get_npm_data and get_github_data into get_data(url) if all you are passing in is a URL with keys etc.